.github/workflows/docker-image.yml

```
@@ -0,0 +1,39 @@
+name: Publish Docker image
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+
+env:
+  IMAGE_NAME: teleprompter
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build image
+        run: docker build . --file Dockerfile --tag $IMAGE_NAME --label "runnumber=${GITHUB_RUN_ID}"
+
+      - name: Log in to registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Push image
+        run: |
+          IMAGE_ID=ghcr.io/${{ github.repository_owner }}/$IMAGE_NAME
+          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+          [ "$VERSION" == "master" ] && VERSION=latest
```

You should be consistent here with brackets usage.

---

The `[[` is because the first line requires [Bash's glob functionality](https://stackoverflow.com/a/12063703) whereas the second line is a basic shell test; they're different functionally, although the second line could use the builtin if you'd prefer?

```suggestion
          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
          [[ "$VERSION" == "master" ]] && VERSION=latest
```

---

IMO unless you 100% know you have a guaranteed need for it, there is no point aiming for POSIX compatibility with your shell scripts. If you put bash into your shebang, use bash to its fullest. So yes, just use `[[ ]]` everywhere. It's modern bash syntax.

---

Sure thing 👍